### PR TITLE
Remove duplicated package names on the list page.

### PR DESCRIPTION
**Pull request by hogo6002**
Fix issues https://github.com/google/osv.dev/issues/1694
Before:
image
After:
image

### CODE REVIEW

File: gcp/appengine/frontend_handlers.py [hunk 1]
```diff
@@ -585,3 +585,24 @@ def package_in_ecosystem(package):
...
+  packages = []
+
+  for affected in vuln_affected:
+    for affected_range in affected['ranges']:
+      if affected_range['type'] in ('ECOSYSTEM', 'SEMVER'):
+        package_entry = affected['package']['ecosystem'] + '/' + affected[
+            'package']['name']
+        if package_entry not in packages:
+          packages.append(package_entry)
+      elif affected_range['repo'] == 'GIT':
```
Comment (another-rex) [inline gcp/appengine/frontend_handlers.py:603]: I think this should still be 'type' right?

File: gcp/appengine/frontend_handlers.py [hunk 2]
```diff
@@ -585,3 +585,24 @@ def package_in_ecosystem(package):
 def osv_has_vuln(vuln_id):
   """Checks if an osv vulnerability exists for the given ID."""
   return osv.Bug.get_by_id(vuln_id)
+
+
+@blueprint.app_template_filter('list_packages')
+def list_packages(vuln_affected: list[dict]):
```
Comment (another-rex) [inline gcp/appengine/frontend_handlers.py:591]: Can you double check if this type is dict? (e.g. printing out the type with `type(vuln_affected)`)

File: gcp/appengine/frontend_handlers.py [hunk 3]
```diff
@@ -585,3 +585,24 @@ def package_in_ecosystem(package):
...
+  """Lists all affected package names without duplicates,
+  remaining the original order."""
```
Comment (another-rex) [inline gcp/appengine/frontend_handlers.py:593]:
```suggestion
  remaining in the original order."""
```

**Review by another-rex — COMMENTED** [review]
Thanks! added some minor comments.

File: gcp/appengine/frontend_handlers.py (hunk 2 above)
Comment (hogo6002) [reply gcp/appengine/frontend_handlers.py:591]: type(vuln_affected) shows list, and type(affected) shows dict. So I think vulnerability.affected is a list[dict].

File: gcp/appengine/frontend_handlers.py (hunk 1 above)
Comment (hogo6002) [reply gcp/appengine/frontend_handlers.py:603]: Thanks!

File: gcp/appengine/frontend_handlers.py (hunk 2 above)
Comment (another-rex) [reply gcp/appengine/frontend_handlers.py:591]: ah cool, thanks!

**Review by another-rex — APPROVED** [review]
LGTM!

**Pull request merged**